packages/@lwc/module-resolver/README.md

```diff
@@ -21,10 +21,156 @@ const result = resolveModule('x/foo', './index.js');
+
+-   [Alias module record](#alias-module-record): Define a file path where an LWC module can be resolved.
+-   [Directory module record](#directory-module-record): Defines a folder path where LWC modules can be resolved.
+-   [NPM package module record](#npm-package-module-record): Defines an NPM package that will expose one or more LWC modules.
+
+```json
+// lwc.config.json
+{
+    "modules": [
+        {
+            "name": "ui/button",
+            "path": "src/modules/ui/button/button.js"
+        },
+        {
+            "dir": "src/modules"
+        },
+        {
+            "npm": "@ui/components"
+        }
+    ]
+}
+```
+
+### Module record types
+
+#### Alias module record
+
+An alias module record maps a module specifier to a file path. An alias module record is defined by two keys:
+
+-   `name` (string, required): The LWC module specifier.
+-   `path` (string, required): The path file path to resolve.
+
+In this example, the `ui/button` LWC module specifier can be resolved from the `src/modules/ui/button/button.js` path.
+
+```json
+{
+    "modules": [
+        {
+            "name": "ui/button",
+            "path": "src/modules/ui/button/button.js"
+        }
+    ]
+}
+```
+
+#### Directory module record
+
+A directory module record specifies a folder path where LWC modules can be resolved from. A directory module record is defined by one key:
+
+-   `dir` (string, required): The directory path containing the modules.
+
+```json
+{
+    "modules": [
+        {
+            "dir": "src/modules"
+        }
+    ]
+}
+```
+
+The directory module record following uses an opinionated folder structure to resolve LWC modules. The directory path can contain one or multiple folders representing the LWC modules `namespace`. Each of those namespace folders can contain one or multiple directories representing the different LWC modules part of this namespace. The name of the directory indicates the LWC module `name`. For a module to be resolved, the LWC module folder should have a file matching the LWC module name, this file is the LWC module entry point.
+
+In this example, if the `dir` key is set `src/modules`, the following LWC modules can be resolved: `ui/button`, `ui/icons`, `shared/utils`.
+
+```
+src
+└── modules/
+    ├── ui/
+    │   ├── button/
+    │   │   ├── button.js
+    │   │   └── button.html
+    │   └── icon/
+    │       └── icon.js
+    └── shared/
+        └── utils/
+            └── utils.js
+```
+
+#### NPM package module record
+
+An NPM package module record tells the resolver than a given NPM package exposes LWC modules than can be resolved. An NPM package module record is defined by one key:
+
+-   `npm` (string, required): The NPM package name exposing the LWC modules.
+
+In this example, the resolver is told to look into the `@ui/components` npm package to look up for LWC modules.
```

Comment:
```suggestion
In this example, the resolver is told to look into the `@ui/components` npm package to look up LWC modules.
```

Reply:
This directive requires the NPM package to follow the _Module resolution_ configuration specified above, right? That is, the NPM package needs either an `lwc.config.json` (at the root of the package, right?) or the `lwc` key in its `package.json`.

Plus it needs to follow the _Exposing LWC modules via NPM packages_ guidance below, right?

Assuming the above is correct, this section needs clarification to make it more obvious how things work.

Reply:
I added more details.